packages/integration/src/lib/catalog/catalog-library-tool/catalog-library-tool.component.ts

```diff
@@ -92,4 +115,220 @@ export class CatalogLibraryToolComponent implements OnInit {
+                context: '',
+                dataDescription: this.getDescription(layer)
+              });
+              catalogRank++;
+            });
+
+            groupsToProcess.forEach((group) => {
+              group.items.forEach((layer) => {
+                const layerToProcess = layer as CatalogItemLayer;
+                const infos = getInfoFromSourceOptions(
+                  layerToProcess.options.sourceOptions,
+                  layer.id
+                );
+                csvOutputs.push({
+                  id: infos.id,
+                  rank: catalogRank.toString(),
+                  layerTitle: layerToProcess.title,
+                  layerGroup: group.title,
+                  catalog: catalog.title,
+                  provider: layerToProcess.externalProvider
+                    ? t.instant('igo.integration.catalog.csv.external')
+                    : t.instant('igo.integration.catalog.csv.internal'),
+                  url: infos.url,
+                  layerName: infos.layerName,
+                  context: '',
+                  dataDescription: this.getDescription(layerToProcess)
+                });
+                catalogRank++;
+              });
+            });
+          });
+
+          csvOutputs = this.matchLayersWithLayersContext(
+            csvOutputs,
+            layerInfosFromDetailedContexts
+          );
+
+          this.downloadCsv(csvOutputs);
+        }
+      );
+  }
+
+  /**
+   * Match a list of layer info with an other list derived from contexts
+   * @param csvOutputs The row list to be written into a csv file
+   * @param layerInfosFromDetailedContexts Layers info derived from the context
+   * @returns An altered list, with layer/context association
+   */
+
+  private matchLayersWithLayersContext(
+    csvOutputs: CsvOutput[],
+    layerInfosFromDetailedContexts: InfoFromSourceOptions[]
+  ): CsvOutput[] {
+    const sep = this.languageService.getLanguage() === 'fr' ? ',' : ';';
+    csvOutputs
+      .filter((b) => b.id !== 'csvHeader')
+      .forEach((csvOutput) => {
+        const matchingLayersFromContext = layerInfosFromDetailedContexts
+          .filter(
+            (l) =>
+              l.id === csvOutput.id ||
+              (l.layerName === csvOutput.layerName && l.url === csvOutput.url)
+          )
+          .map((f) => f.context);
+        csvOutput.context = matchingLayersFromContext.join(sep);
+      });
+    return csvOutputs;
+  }
+
+  /**
+   * Write a CSV file, the column separator is based on language.
+   * @param csvOutputs The row list to be written into a csv file
+   */
+  private downloadCsv(csvOutput: CsvOutput[]) {
+    const sep = this.languageService.getLanguage() === 'fr' ? ';' : ',';
+    let csvContent = csvOutput
+      .map((e) => {
+        return `${e.rank}${sep}${e.layerTitle}${sep}${e.layerGroup}${sep}${e.catalog}${sep}${e.provider}${sep}${e.url}${sep}${e.layerName}${sep}${e.context}${sep}${e.dataDescription}`;
```

Comment:
Pourquoi ne pas afficher le id?
Si le id est toujours en premier et qu'on ne veut pas l'afficher, on pourrait simplifier:
```suggestion
        return Object.values(e).shift().join(sep);
```

Reply:
Je crain qu'en prenant ceci, on mélange l'ordre des champs dans le csv.